clients/fides-js/src/lib/consent.tsx

```diff
@@ -112,6 +126,49 @@ const getLocation = async (
   }
 };
 
+/**
+ * Hide pre-existing link in the DOM if we do not need to trigger a modal
+ */
+const hideModalLink = (debug: boolean): void => {
+  // TODO- it's possible that this element does not exist by the time this method runs
+  const modalLinkEl: HTMLElement | null =
+    document.getElementById(FIDES_MODAL_LINK);
+  if (modalLinkEl) {
+    debugLog(debug, "modal link element exists, attempting to hide it");
+    // TODO: hide link
+    // eslint-disable-next-line no-param-reassign
+    modalLinkEl.style.display = "none";
+  }
+  debugLog(
+    debug,
+    "modal link element does not exist, so there is nothing to hide"
+  );
+};
+
+/**
+ * Update the pre-existing modal link in the DOM to trigger the modal
+ */
+const bindModalLinkToModal = (debug: boolean): void => {
+  // TODO- it's possible that this element does not exist by the time this method runs
+  const modalLinkEl: HTMLElement | null =
+    document.getElementById(FIDES_MODAL_LINK);
+  if (
+    modalLinkEl &&
+    // TODO: does this need to be an HTMLAnchorElement?
+    modalLinkEl instanceof HTMLAnchorElement
```

Comment:
actually is that true? I thought we sometimes wanted the link to go to the privacy center, but is it the case we always want to open the modal?

Reply:
After speaking with @NevilleS last week, we decided the best path forward would be to have the customer hard-code 2 links onto their site: 
1. The link to the PC- this link will not depend on experience config, and will always exist. This is because DSRs will still need to be processed whether or not notices require consent.
2. The link that opens the consent modal- we believe the safest way to do this is to hide the link from the DOM if there are no notices that apply OR trigger the modal from the link if there are applicable notices.

So, from what I'm hearing, it sounds like this modal link element does not need to be an HTMLAnchorElement. From a UX perspective, I believe this element is intended to appear like a link and not a button. Can @mfbrown confirm whether or not we should we allow the customer to use any element they like (including link or button or `a` tag), as long as the id on that element matches what we expect?

@allisonking - from a DOM manipulation standpoint, it shouldn't be more difficult for us if a customer were to use a button vs link vs anchor element, right?

Reply:
thanks for the clarification, that helps! yes, from a DOM manipulation standpoint, we can configure an `onClick` for either an anchor or a button. I think it's weird to have an anchor have an onClick, but it might be just fine. 

I'm not sure how we will be able to trigger the modal from this manipulated link, though. we might have to refactor some things, but we can discuss that after I get https://github.com/ethyca/fides/pull/3291 into a better state